addons/l10n_latam_check/models/account_payment.py

```diff
@@ -0,0 +1,379 @@
+import stdnum
+from odoo import fields, models, _, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.misc import format_date
+from odoo.osv import expression
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class AccountPayment(models.Model):
+    _inherit = 'account.payment'
+
+    # Third party check operation links
+    l10n_latam_check_id = fields.Many2one(
+        comodel_name='account.payment',
+        string='Check',
+        readonly=True, states={'draft': [('readonly', False)]},
+        copy=False,
+    )
+    l10n_latam_check_operation_ids = fields.One2many(
+        comodel_name='account.payment',
+        inverse_name='l10n_latam_check_id',
+        string='Check Operations',
+        readonly=True,
+    )
+    l10n_latam_check_current_journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string="Check Current Journal",
+        compute='_compute_l10n_latam_check_current_journal', store=True,
+    )
+    # Warning message in case of unlogical third party check operations
+    l10n_latam_check_warning_msg = fields.Text(
+        compute='_compute_l10n_latam_check_warning_msg',
+    )
+    l10n_latam_check_number = fields.Char(
+        compute='_compute_l10n_latam_check_number', inverse='_inverse_l10n_latam_check_number',
+    )
+    # New third party check info
+    l10n_latam_check_bank_id = fields.Many2one(
+        comodel_name='res.bank',
+        string='Check Bank',
+        compute='_compute_l10n_latam_check_data', store=True, readonly=False,
+        states={'posted': [('readonly', True)], 'cancel': [('readonly', True)]},
+    )
+    l10n_latam_check_issuer_vat = fields.Char(
+        string='Check Issuer VAT',
+        compute='_compute_l10n_latam_check_data', store=True, readonly=False,
+        states={'posted': [('readonly', True)], 'cancel': [('readonly', True)]},
+    )
+    l10n_latam_check_payment_date = fields.Date(
+        string='Check Payment Date',
+        readonly=True, states={'draft': [('readonly', False)]},
+    )
```

Comment:
Checks have an expiry date. (otherwise people would keep using checks instead of money)

Reply:
> Check Payment Date

It doesn't say "expiry date"

Reply:
@william-andre it is not an expiration date but it is the date when you can collect the payment/cash (execute the check and collect the money to your bank), so it is better to say Check Payment Date

Reply:
If you want something more understandable in ENG it could be "Collection Date" or something similar to that.
As that date determines from when you can make that check into cash

Reply:
Some other options for the name:
1. Collection Date
2. Cash-in Date

@william-andre Probably the 2nd one is a better match, as it refers to the date that the money can be collected and is more used by banks, please pick the one that better makes sense for you and we will add it.

Furthermore, let's add a tooltip (helper) to that field, which could be:
* Date from when you can cash in the check, turn the check into cash

Or something like that.
Once this is decided, @zaoral can you help me add the helper + change the name?

Thanks!
cc' @jco-odoo @jjscarafia